### DISCUSSION TITLE
Exception: адрес недоступен

### DISCUSSION THREAD

**Discussion by amd64fox** [Chimildic/goofy#27] [Помощь с ошибкой] [answered]
Такая вот ошибка останавливает скрипт. 
```
16 янв. 2021 г., 14:04:00	Ошибка	Exception: Адрес недоступен: https://api.spotify.com/v1/tracks/?ids=3JcHh42KnYA3ypJl3F2C5Z,4hC7Sl95ZNteCDGmHWCZCL,7gCNoI25PFFbUdYJneYYWC,39hH92CGKxiF9sNKW5HXIm,43MOLFEnFOOSn65XJy6Gs4,3ZwZN6blYamlz1SwtTcZYX,4tEv80D8JS4tpVWeFjc3xn,5oyCIQWuSgFB25yA66AIcG,5Uuv1Ro9jh49s8QW9fD5dr,6KH3n3ZjCcqvjt3SK7e0xW,616x16Jz4OZevroEYf2HYl,62Mx4j3cybbW4aXhogbtH4,6WDMOh0LJY9EHO4aIE7dPh,774kFk9YCPyTYR7sSIDqum,1VWS04zCRLVLsngu3RZ7Ez,1ULM90V9nWEBqoy51BPqzn,7yn0Hp7mHjbHkBLdGQU287,74VR3AkGPhbYXnxcOYa16x,6lTTzSk1hRrxp4VMwXBp2l,0Dz0A85c0hQWvkyCB2hlYq,5i5zoffpGBeRHIbwFxmydu,5YYmzeLgYiitbrs7Z2AdMj,409WAJPEw4a73vs2c5FIZI,5PKtqiUwkjveVYzmj4gMnz,7cQkNSSjdyU1Aw5RVbZZhd,3zw9Hhzlgg0QqSvXcmKJxF,2VR7EqYSBdqPVW5u3gcotK,5i2MXxinpfJCj1ynihiNnt,72yD9FdqJPuZEJEHlzi4TW,1g8lVIsEkDaGD1nKw6fwUb,10nOWGyxyXW42t9NH6BNSS,2W2eaLVKv9NObcLXlYRZZo,79DHdvI3WjP9DGmJL0upxi,5m1QIZfukebnjU2TpdvWUO,2Yidr7pOJxWjsbdAxYU4FY,0UHxkkgI5GfV0XBRA1rqfe,4TmDmB5lpAtv6rmpUCDjBy,4Jij1Pvde2uXUHZbA1uM5o,7C1nBJzb8fiSIraXh61Q6P,2dJ4B4AFU8BzwZRzdvkTv8,2me5dUmfgVwk3iw3wHdTza,5zHPOglU6vWzaaJJRN7c7z,7AMEPQcRRojeoBWJkD5wcU,52csNdG1mOCeGLeUHr7F6l,1ffhRcMHiQpvtCWJLYkPkE,3COUNGSSGOy6MD3FtHLvAl,3m8UfZwPMMbiWfGR8Lg2cR,5HiAN0PZVXUSMgpiqrScC9,2u1FWVxAb16qbgwPgygAdj,6g6qt1tnnW5S1sCwAWnW5m
    at sendPack(Library:71:35)
    at fetchAll(Library:58:31)
    at getAll(Library:2392:34)
    at getFullObjByIds(Library:2371:9)
    at getTracks(HistoryManager:13:31)
    at History(Main:3:37)
```

3-ая строка из Main
`let tracks_new = HistoryManager.getTracks();`

13-ая строка из HistoryManager
`return SpotifyRequest.getFullObjByIds('tracks', readTrackIds(), 50);`

**Comment by amd64fox** [comment]
Вот еще похожая ошибка в другом скрипте
```
Exception: Адрес недоступен: https://api.spotify.com/v1/search/?q=babymetal%20bxmxc&type=track&limit=1
    at sendPack(Library:71:35)
    at fetchAll(Library:58:31)
    at getAll(Library:2392:34)
    at multiBestSearch(Library:2103:31)
    at multisearch(Library:2082:28)
    at multisearchTracks(Library:2063:16)
    at getTracks(Library:1828:23)
    at Moi_Mix(Main:509:25)
```

**Comment by Chimildic** [comment, marked as answer]
Посмотрел [трекер](https://issuetracker.google.com/issues?q=componentid:191640%2B%20Exception:%20Address%20unavailable) по этой ошибке. Она связана с блокировкой запроса. Только кем? (Google, Spotify, или вообще провайдером). Внятного объяснения нет. Грубо говоря, гугл для отправки запроса выделяет нам свой ресурс. И этот ресурс на каком-то этапе натыкается на блокировку, а мы получаем неинформативное исключение.

Стектрейс показывает, что это дважды случилось в `fetchAll`. Он отправляет много запросов параллельно. Отсюда чем больше ресурса/запросов, тем вероятнее наткнуться на плохой/блокируемый ресурс.

Попробую добавить перехват исключения, через паузу повторить отправку запросов. И попытаюсь вернуть то, что уже было получено.

Если ошибка происходит часто, снизьте значение [параметра](https://chimildic.github.io/goofy/#/guide?id=Параметры) `REQUESTS_IN_ROW`, например, до 30. Не забудьте запустить функцию `setProperties`, чтобы значение обновилось.

**Reply by amd64fox** [reply]
У меня после этого репорта больше подобного не было, и `REQUESTS_IN_ROW` тоже не менял, стоит по дефолту. Мб действительно это провайдер на тот момент что-то делал.

**Reply by Chimildic** [reply]
Если что, имею ввиду не локально вашего провайдера, а тех, кто ресурсы гугла обслуживает.

**Comment by Chimildic** [comment]
В версии 1.4.2 предпринята попытка продолжить исполнение кода при получении данного исключения.
